File: jarvis-plc-integration/src/main/java/com/mikhalochkin/jarvis/plc/integration/api/PlcClient.java
package com.mikhalochkin.jarvis.plc.integration.api;

import java.util.Map;

/**
 * Client for programmable logic controller.
 *
 * @author Alex Mikhalochkin
 */
public interface PlcClient {

    /**
     * Turns on the specified port.
     *
     * @param portNumber port number.
     */
    void turnOn(int portNumber);

    /**
     * Turns off the specified port.
     *
     * @param portNumber port number.
     */
    void turnOff(int portNumber);

    void turnOffAll();

    void turnOnAll();

    /**
     * Returns statuses of all ports.
     *
     * @return ports' statuses.
     */
    Map<Integer, Boolean> getOutPortsStatuses();//TODO: change return type
}

File: src/main/kotlin/com/mega/demo/integration/api/PlcClient.kt
package com.mega.demo.integration.api

/**
 * Http client for programmable logic controller.
 *
 * @author Alex Mikhalochkin
 */
interface PlcClient {

    fun turnOn(port: Int)

    fun turnOff(port: Int)

    fun getPortStatuses(): Map<Int, Boolean>
}
